slick/src/main/scala/za/co/absa/fadb/slick/SlickQuery.scala

```diff
@@ -16,14 +16,64 @@
 
 package za.co.absa.fadb.slick
 
-import slick.jdbc.{GetResult, SQLActionBuilder}
-import za.co.absa.fadb.Query
+import slick.jdbc.{GetResult, PositionedResult, SQLActionBuilder}
+import za.co.absa.fadb.exceptions.StatusException
+import za.co.absa.fadb.status.FunctionStatus
+import za.co.absa.fadb.{FunctionStatusWithData, Query, QueryWithStatus}
 
 /**
-  * SQL query representation for Slick
-  * @param sql        - the SQL query in Slick format
-  * @param getResult  - the converting function, that converts the [[slick.jdbc.PositionedResult slick.PositionedResult]] (the result of Slick
-  *                   execution) into the desire `R` type
-  * @tparam R         - the return type of the query
-  */
+ *  SQL query representation for Slick
+ *  @param sql        - the SQL query in Slick format
+ *  @param getResult  - the converting function, that converts the [[slick.jdbc.PositionedResult slick.PositionedResult]] (the result of Slick
+ *                   execution) into the desire `R` type
+ *  @tparam R         - the return type of the query
+ */
 class SlickQuery[R](val sql: SQLActionBuilder, val getResult: GetResult[R]) extends Query[R]
+
+/**
+ *  SQL query representation for Slick with status
+ *  @param sql        - the SQL query in Slick format
+ *  @param getResult  - the converting function, that converts the [[slick.jdbc.PositionedResult slick.PositionedResult]] (the result of Slick
+ *                   execution) into the desire `R` type
+ *  @tparam R         - the return type of the query
+ */
+class SlickQueryWithStatus[R](
+  val sql: SQLActionBuilder,
+  val getResult: GetResult[R],
+  checkStatus: FunctionStatusWithData[PositionedResult] => Either[StatusException, PositionedResult]
+) extends QueryWithStatus[PositionedResult, PositionedResult, R] {
+
+  /*
+   * Processes the status of the query and returns the status with data
+   * @param initialResult - the initial result of the query
+   * @return the status with data
+   */
+  override def processStatus(initialResult: PositionedResult): FunctionStatusWithData[PositionedResult] = {
+    val status: Int = initialResult.<<
+    val statusText: String = initialResult.<<
+    FunctionStatusWithData(FunctionStatus(status, statusText), initialResult)
+  }
+
+  /*
```

Comment:
```suggestion
  /**
```

? also on other places, just to have it more colorful and consistent in IntelliJ

Reply:
fixed